c2rust-analyze/src/main.rs

```diff
@@ -449,13 +449,65 @@ fn run(tcx: TyCtxt) {
             assert_eq!(local, l);
         }
 
+        // Find all calls to `malloc`, `calloc`, `realloc`, and `free`
+        // and track their destination locals as being libc::c_void
+        // pointers to special-case downstream casts
+        for bb_data in mir.basic_blocks().iter() {
+            if let Some(term) = &bb_data.terminator {
```

Comment:
```suggestion
        for term in mir.basic_blocks().iter().filter_map(|bb| bb.terminator) {
```

Something like this could help just to reduce the indentation level a bit.

Reply:
will opt out of this change. feel free to suggest an entire working change if you feel strongly about it

Reply:
```suggestion
        for term in mir
            .basic_blocks()
            .iter()
            .filter_map(|bb| bb.terminator.as_ref())
        {
```

This should work (also needs to remove the corresponding `}` and re-format).  Not really that important, but it's helpful for reading to avoid very indented code.